src/sinks/datadog/logs/config.rs

```diff
@@ -97,35 +92,29 @@ impl SinkConfig for DatadogLogsConfig {
 
         let request_settings = self.request.unwrap_with(&TowerRequestConfig::default());
         let client = HttpClient::new(tls_settings, cx.proxy())?;
-        let client = ServiceBuilder::new()
-            .check_service::<HttpClient, Request<Body>, Response<Body>, HttpError>()
+        let healthcheck =
+            healthcheck(client.clone(), self.get_uri(), self.default_api_key.clone()).boxed();
+
+        let mut client = ServiceBuilder::new()
             .rate_limit(
                 request_settings.rate_limit_num,
                 request_settings.rate_limit_duration,
             )
-            .check_service::<HttpClient, Request<Body>, Response<Body>, HttpError>()
-            .retry(LogApiRetry)
-            .check_service::<HttpClient, Request<Body>, Response<Body>, HttpError>()
-            // TODO introduction causes a compilation failure for want of a Sync
-            // implementation on the `State` enum of adapative concurrency.
+            // Ideally we'd use ARC but the type constraints don't quite line
+            // up. We'll address that in a follow-up pull request.
```

Comment:
Is this something we think may block the work to have ARC on by default next release?

Reply:
It would either imply that ARC would not be on by default for this sink -- assuming that passes review -- or this work gets held up while we work out the integration.

Reply:
Yeah, I was actually going to bring this up too. I hesitate to introduce a regression in behavior. At the least it'd be good to fix this before the next release (0.17.0), but even regressing the nightlies is not ideal. My preference would be to address this before this PR goes in.

Maybe @bruceg has thoughts about how it could be wired up.

Reply:
I would also prefer this to be drop-in, for what it's worth. 

The major question that needs to be resolved is what we want to do about the retry logic. Tower's setup uses a [`Policy`](https://docs.rs/tower/0.4.8/tower/retry/trait.Policy.html) for the retry layer and our ARC uses a `RetryLogic`. At least in this sink they'll end up being the same implementation, just in different traits. Do we need two similar traits in the project or can we slim down to one?